lib/fluent/plugin_helper/metrics.rb

```diff
@@ -42,7 +42,11 @@ def configure(conf)
         @plugin_type_or_id = if self.plugin_id_configured?
                                self.plugin_id
                              else
-                               "#{conf["@type"] || conf["type"]}.#{self.plugin_id}"
+                               if type = (conf["@type"] || conf["type"])
+                                 "#{type}.#{self.plugin_id}"
+                               else
+                                 "#{self.class.to_s.split("::").last.downcase}.#{self.plugin_id}"
```

Comment:
This is for `out_file`'s default buffer plugin settings:
https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/out_file.rb#L61-L65